docs/odcs.rst

```diff
@@ -24,84 +24,214 @@ idea is to not require the user to provide a yum repo url when building
+to contain a *compose* section. This section will include all the required
 information for requesting a compose to be created by ODCS::
 
     compose:
-      # Optional, default False. Inherit compose used for building parent image
-      inherit: true
-      # Optional, default "tag".
-      type: tag
-      # Required for "tag" type. Must contain at least one item.
+      # Required for ODCS Koji "tag" type usage. Must contain at least one item.
       packages:
       - package-1-name # This is a package name, not an NVR.
       - package-2-name
-      # Optional. Default value is configured in OSBS environment
-      sigkeys:
-      - AB123456
+      # Optional. Default and possible values are configured in OSBS
+      # environment.
+      signing_intent: release
 
 Alternatively, for modules support::
 
     compose:
-      type: module
       # Eventually, ODCS will support ":" separator, for now "-" is used:
       #     https://pagure.io/odcs/issue/98
-      # Required for "module" type. Must contain at least one item.
+      # Required for ODCS "module" type usage. Must contain at least one item.
       modules:
       - "module_name1:stream1"
       - "module_name2:stream1"
 
+**Exactly one non-empty list of "modules" or "packages" must be provided. If
+both or none are defined build will fail.**
+
 Behavior Changes
 ----------------
 
-During an OSBS build, if ``container.yaml`` exists **and** ``compose`` key is defined:
+During an OSBS build, if ``container.yaml`` exists **and** *compose* key is
+defined:
 
 - OSBS orchestrator build will request a compose from ODCS based on specified
-  parameters in ``compose`` section. Once compose completes, build resumes and
-  the URL for the generated compose is passed to worker builds via
-  ``yum_repourls`` build parameter. *NOTE: If odcs-signed parameter was set to
-  false, the compose request will ignore the sigkeys provided in
-  container.yaml.*
-
-- If ``compose`` section sets ``inherit`` to ``true``, the ODCS composes used in
-  building the parent image will be re-used. This may require OSBS to request
-  compose to be regenerated if it’s in *remove* state. Once compose is
-  available, build resumes and the URL for the generated compose is passed to
-  worker builds via ``yum_repourls`` build parameter. *NOTE: this can be used in
-  conjunction with the method stated on previous bullet point. In which case,
-  multiple ODCS composes are used.*
+  parameters in *compose* section. Once compose completes, build resumes and
+  the compose ID is passed to the worker builds via the new ``compose_ids``
+  build parameter.
 
 - In the Koji build metadata, ``build.extra.image.odcs.compose_ids`` will be a
   list of each ODCS compose used. Additionally,
-  ``build.extra.image.odcs.signed`` will correspond to the value of
-  ``odcs_signed`` build parameter which defaults to ``True``
+  ``build.extra.image.odcs.signing_intent`` will correspond to the signing
+  intent of the ODCS composes.
```

Comment:
Also mention `build.extra.image.odcs.signing_intent_overridden` here?

Reply:
Good idea. Changed this paragraph to this format:

> In the Koji build metadata:
> - build.extra.image.odcs.compose_ids: list of each ODCS compose used.
> - build.extra.image.odcs.signing_intent: lest restrictive signing intent of the ODCS composes.
> - build.extra.image.odcs.signing_intent_overridden: whether or not the signing intent used is different than the one defined in container.yaml.

Reply:
Can signing_intent_overridden be absent, and if so should it be assumed to be false?

Reply:
I'm thinking it's better to always have it in there. Makes consuming this data easier.